### PR TITLE
MIME + PTML Fixes

### PR DESCRIPTION
**Pull request by mrjackv**

### CODE REVIEW

**Comment by aleclearmind** [comment]
The branch name doesn't seem appropriate.

**Comment by aleclearmind** [comment]
Yes, branch names are important.

**Comment by mrjackv** [comment]
Split the PR, since the changes where uncorrelated
This is for some fixes to the PTML output and the new MIME convention

**Review by aleclearmind — CHANGES_REQUESTED** [review]

File: include/revng/Pipes/Ranks.h [hunk 1]
```diff
@@ -22,7 +22,11 @@ inline auto BasicBlock = defineRank<"basic-block", MetaAddress>(Function);
 inline auto Instruction = defineRank<"instruction", MetaAddress>(BasicBlock);
 
 inline auto Type = defineRank<"type", model::Type::Key>(Binary);
-inline auto TypeField = defineRank<"type-field", std::tuple<uint64_t>>(Type);
+using SFK = model::StructField::Key;
+using UFK = model::UnionField::Key;
```
[the remark this answers is not in the document]
Comment (mrjackv) [reply include/revng/Pipes/Ranks.h]: Done

File: include/revng/Yield/Pipes/YieldControlFlow.h [hunk 2]
```diff
@@ -16,28 +16,22 @@
...
   &kinds::FunctionAssemblyInternal,
-  FunctionAssemblyName,
-  FunctionAssemblyMIMEType>;
+  FunctionAssemblyYamlName,
+  FunctionAssemblyYamlMIMEType>;
 
-inline constexpr char FunctionAssemblyPTMLMIMEType[] = "application/"
-                                                       "x.yaml.function-"
-                                                       "assembly.ptml-body";
+inline constexpr char FunctionAssemblyPTMLMIMEType[] = "text/x.asm+ptml+yaml";
```
[the remark this answers is not in the document]
Comment (mrjackv) [reply include/revng/Yield/Pipes/YieldControlFlow.h:26]: Appended section about MIME types in the PTML pad

**Comment by aleclearmind** [comment]
Merged in 025f6d5. Thanks!

**Pull request closed without merging**